packages/library-legacy/src/connection.ts

```diff
@@ -3212,23 +3312,25 @@ export class Connection {
     commitmentOrConfig?: Commitment | GetBalanceConfig,
   ): Promise<RpcResponseAndContext<number>> {
     /** @internal */
-    const {commitment, config} =
-      extractCommitmentFromConfig(commitmentOrConfig);
-    const args = this._buildArgs(
-      [publicKey.toBase58()],
-      commitment,
-      undefined /* encoding */,
-      config,
+    const params = buildParams<'getBalance'>(
+      [publicKey.toBase58() as Base58EncodedAddress],
+      commitmentOrConfig,
+      {
+        commitment: this._commitment,
+      },
     );
-    const unsafeRes = await this._rpcRequest('getBalance', args);
-    const res = create(unsafeRes, jsonRpcResultAndContext(number()));
-    if ('error' in res) {
-      throw new SolanaJSONRPCError(
-        res.error,
-        `failed to get balance for ${publicKey.toBase58()}`,
-      );
-    }
-    return res.result;
+    return await this._solanaRpc
+      .getBalance(...params)
+      .send()
+      .then(unsafeRes =>
+        create(unsafeRes, jsonRpcResultAndContextExperimental(number())),
+      )
+      .catch(e => {
+        throw new SolanaJSONRPCError(
+          e,
+          `failed to get balance for ${publicKey.toBase58()}`,
+        );
+      });
```

Comment:
You can write this more like this:

```ts
try {
    const unsafeRes = await this._solanaRpc.getBalance(...params).send();
    return create(unsafeRes, jsonRpcResultAndContextExperimental(number()));
} catch (e) {
    throw new SolanaJSONRPCError(e, `failed to get balance for ${publicKey.toBase58()}`);
}
```

Reply:
This was kinda how they were written beforehand. Now that I'm looking at it, I'm assuming this would be better because it catches errors on the RPC request and response coercion, rather than just request?

Reply:
Uhhhh… hmm. Good question. I guess you probably _just_ want to wrap the rpc request in the try/catch so as to minimize the amount of change here (response coercer would throw straight up in both cases)

```ts
let unsafeRes;
try {
  unsafeRes = await this._solanaRpc.getBalance(...params).send();
} catch (e) {
  throw new SolanaJSONRPCError(e, `failed to get balance for ${publicKey.toBase58()}`); 
}
return create(unsafeRes, jsonRpcResultAndContextExperimental(number()));
```

Reply:
The _style_ is up to you (I prefer try/catch style, but the other is fine) but the way you had it written above, `SolanaJSONRPCError` will get thrown on _both_ RPC errors _and_ failures to do runtime typechecking in `create`, which diverges from the current behavior and was maybe hard to spot.

Reply:
Makes sense, thanks! I sort of realized after I asked what you were driving at